### DISCUSSION TITLE
Fold commands

### DISCUSSION THREAD

**Discussion by mrbjarksen** [nvim-neo-tree/neo-tree.nvim#368] [Show and tell]
I've been tinkering around with creating commands and mappings which behave like Vim's default fold mappings. My custom commands can be found in [this GitHub Gist](https://gist.github.com/mrbjarksen/ac8b797233ef38eef89d6adfaebef59e). I use them with the following mappings:
```lua
['zo'] = commands.open_fold,
['zO'] = commands.open_folds_rec,
['zc'] = commands.close_fold,
['zC'] = commands.close_folds_rec,
['za'] = commands.toggle_fold,
['zA'] = commands.toggle_folds_rec,
['zv'] = commands.fold_view_cursor,
['zM'] = commands.close_all_folds,
['zR'] = commands.expand_all_folds,
['[z'] = commands.focus_fold_start,
[']z'] = commands.focus_fold_end,
['zj'] = commands.focus_next_fold_start,
['zk'] = commands.focus_prev_fold_end,
```
I'm pretty happy with the results. There are some mappings missing, though, most notably the ones which manipulate `'foldenable'` and `'foldlevel'`. I've decided this is as much as I can stomach, however. I'm curious if anyone here has any ideas for the remaining mappings or improvements on my commands (I suspect there is a better way to recursively scan files, for example).

Edit: I noticed a bug that throws errors when opening different Neo-tree sources on the same side as the current one. This should be fixed now, at the cost of the saved cursor node (used for folding) being forgotten when leaving the Neo-tree window. Of course this isn't the end of the world, but it's not ideal.

**Comment by swarn** [comment]
Ha! Back in July, I hadn't seen this, and I reinvented it. I added my version [to the wiki](https://github.com/nvim-neo-tree/neo-tree.nvim/wiki/Recipes#emulating-vims-fold-commands) at that time. Like you, I don't have a version of `zn` and friends. I also don't have your fold navigations, like `zk`. Very cool!

**Reply by mrbjarksen** [reply]
Oh, cool! I didn't think to add to the wiki at the time, would you mind if I added a link to this discussion next to your implementation?

**Reply by swarn** [reply]
Not at all!

**Reply by mrbjarksen** [reply]
Great!
